scripts/build_scripts/build-adios2-kvcache.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+# This script is to build the remote server with cache enabled.
+# Attention: hiredis cannot be installed by apt-get, since the default version is too old (libhiredis0.14 (= 0.14.1-2)).
+#            We need to build it from source code. You can also use the following scripts to install hiredis (v1.2.0).
+
+# sample usage: in project home directory:
+# source scripts/build_scripts/build-adios2-kvcache.sh --build 
+# source scripts/build_scripts/build-adios2-kvcache.sh --start
+# source scripts/build_scripts/build-adios2-kvcache.sh --stop
+
+if [ -z "${BUILD_DIR}" ]
+then
+    BUILD_DIR="${PWD}"/build
+fi
+
+if [ ! -d "${BUILD_DIR}" ]
+then
+    mkdir -p "${BUILD_DIR}"
+fi
+
+SW_DIR="${BUILD_DIR}"/sw
+if [ ! -d "${SW_DIR}" ]
+then
+    mkdir -p "${SW_DIR}"
+fi
+
+build_cache() {
+    # redis - in-memory data structure store
+    redis_dir="${SW_DIR}"/redis-7.2.3
+    if [ ! -d "${redis_dir}" ]
+    then
+        cd "${SW_DIR}" || exit
+        curl -L -o redis-7.2.3.tar.gz https://github.com/redis/redis/archive/refs/tags/7.2.3.tar.gz
+        tar -xzf redis-7.2.3.tar.gz
+        rm redis-7.2.3.tar.gz
+        cd "${redis_dir}" || exit
+        # cannot accleerate by 'make -j8'. It will cause error.
+        make
+
+        # hiredis - C client library to connect Redis server
+        cd "${redis_dir}"/deps/hiredis || exit
+        mkdir build && cd build || exit
+        cmake .. -DCMAKE_INSTALL_PREFIX="${SW_DIR}"/hiredis
+        make -j32
+        make install
+    fi
+
+    cd "${BUILD_DIR}" || exit
+    cmake .. -DADIOS2_USE_Cache=ON \
+            -DADIOS2_USE_Python=ON \
+            -DCMAKE_PREFIX_PATH="${SW_DIR}/hiredis" \
+            -DCMAKE_INSTALL_PREFIX="${SW_DIR}"/adios2
+
+    make -j32
+    make install
+    cd "${BUILD_DIR}"/../ || exit
+    echo "Build completed."
+    unset BUILD_DIR
+}
+
+start_services() {
```

Comment:
The "DoRemote" environment variable was a quick hack to trigger remote access while we figured out how we really wanted to to it, but I'm a little surprised that it works to have DoRemote set for the adios2_remote_server.  Generally you'd want that set for the clients, not for the server (which runs the same Get() code, but which has to do local access).   If the DoRemote and useKVCache variables are for the clients which follow, maybe move those to the end of this script?  Or is does the server need useKVCache?  (That may be confusion on my part WRT where the caching happens, on the server or on the client.)

Reply:
The current cache design is for clients, and I have moved those two declarations to the end of build process.